### PR TITLE
Feature/issue 39 alumni companies block

### PR DESCRIPTION
**Pull request by SyleimanovSergey**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [ ] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

_Describe the source of requirement, like related issue link._

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [ ] Changes are tested locally

### CODE REVIEW

**Comment by oleksiisydorenko1** [comment]
@SyleimanovSergey Did you contact with designers, because from last conversation I got they wanna make some minor changes. @liliya-trofimova

File: components/AlumniCompanies/alumni-companies.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { onUpdateCompaniesList } from './utils';
+import { AlumniCompaniesP, ArrayAlumniCompanies } from '../../types';
+
+import styles from './styles.module.scss';
+
+const AlumniCompanies: React.FC<AlumniCompaniesP> = props => {
+  const { companies } = props || {};
+
+  const [companiesList, setCompaniesList] = useState<ArrayAlumniCompanies[]>(
+    companies.slice(0, 6)
```
Comment (oleksiisydorenko1) [inline components/AlumniCompanies/alumni-companies.tsx]:
```suggestion
    companies.slice(0, companies.length)
```
why 6 ?)

Comment (aleksKudr) [reply components/AlumniCompanies/alumni-companies.tsx]: Maybe create limit variables?

File: components/AlumniCompanies/alumni-companies.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        We are immensely proud of RS School alumni who build their successful
+        carriers in ambitious IT companies
+      </p>
+      <div className={styles.imagesSection}>
+        {companiesList.map(({ name, picture, _key: key }, index) => (
+          <SwitchTransition mode={'out-in'} key={index}>
+            <CSSTransition
+              key={key}
+              unmountOnExit
+              timeout={{ enter: 1200, exit: 1200, appear: 1000 }}
```
Comment (aleksKudr) [inline components/AlumniCompanies/alumni-companies.tsx]: Please remove this from component. Use variables. Maybe with global const or useMemo

File: components/AlumniCompanies/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  background-color: #191b1d;
+
+  &:after {
+    display: block;
+    content: '';
+
+    height: 100%;
+    width: 5px;
+
+    background-color: #fff;
```
Comment (aleksKudr) [inline components/AlumniCompanies/styles.module.scss]: bg-white

File: components/AlumniCompanies/alumni-companies.tsx (hunk 1 above)
Comment (SyleimanovSergey) [reply components/AlumniCompanies/alumni-companies.tsx]: This is the number of items displayed. Corrected in the new commit

File: components/AlumniCompanies/alumni-companies.tsx (hunk 2 above)
Comment (SyleimanovSergey) [reply components/AlumniCompanies/alumni-companies.tsx]: I hope I got it right. Updated in the new commit

File: components/AlumniCompanies/styles.module.scss (hunk 3 above)
Comment (SyleimanovSergey) [reply components/AlumniCompanies/styles.module.scss]: Ready

File: types/alumni-companies.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { SanityImageObject } from '@sanity/image-url/lib/types/types';
+
+export type ArrayAlumniCompanies = {
```
Comment (oleksiisydorenko1) [inline types/alumni-companies.ts]: ArrayAlumniCompanies => AlumniCompanies

File: types/alumni-companies.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  _key: string;
+  name: string;
+  picture: SanityImageObject;
+};
+
+export type AlumniCompaniesFetchT = {
+  _id: string;
+  companies: Array<ArrayAlumniCompanies>;
```
Comment (oleksiisydorenko1) [inline types/alumni-companies.ts]: Array<ArrayAlumn , let's avoid duplication :)

Comment (oleksiisydorenko1) [reply types/alumni-companies.ts]: Also, companies should contain Array<AlumniCompany>, not companies (because it's already array of companies)

Comment (oleksiisydorenko1) [reply types/alumni-companies.ts]: `Array<AlumniCompany>;`

**Review by oleksiisydorenko1 — APPROVED** [review]

**Review by aleksKudr — APPROVED** [review]

**Comment by oleksiisydorenko1** [comment]
@SyleimanovSergey are we merging this?

**Comment by SyleimanovSergey** [comment]
@oleksiisydorenko1 > 
I still have some work to do there, so I'll merge it by the end of the year.

**Comment by oleksiisydorenko1** [comment]
> @oleksiisydorenko1 > I still have some work to do there, so I'll merge it by the end of the year.

Got you, so Happy New Year)))

**Pull request merged**